### PR TITLE
Adding a list of translated texts, to check the right path

### PR DESCRIPTION
**Pull request by sharifrezvani**
I just translated a couple of sentences and texts to see if I'm on the right path or not. 

This is related to https://github.com/keycloak/keycloak/issues/19604

Thank you guys.

### CODE REVIEW

**Comment by jonkoops** [comment]
Please also refrain from updating the PR with `main` through the GitHub interface. This is not needed for a merge and results in unnecessary merge commits and notifications.

**Review by jonkoops — CHANGES_REQUESTED** [review]
This is a good change, but to add a locale properly it also needs to be added [under the base theme](https://github.com/keycloak/keycloak/tree/main/themes/src/main/resources-community/theme/base/admin/messages). Create a file there that contains the same labels as the [English version](https://github.com/keycloak/keycloak/blob/main/themes/src/main/resources/theme/base/admin/messages/messages_en.properties).

This will allow Keycloak to pick up the locale as expected.

**Comment by sharifrezvani** [comment]
@jonkoops Thank you for your guidance. I believe that in my latest commit I have created all the necessary files and all of the needed translations. Let me know if there's anything that I would need to change.

**Comment by sharifrezvani** [comment]
@jonkoops Just as a reminder, is there anything else that I need to do or are the requested changes satisfied now?

**Comment by jonkoops** [comment]
@sharifrezvani Sorry for the late reply, I was busy with some other tasks. In order for Keycloak to pick up these new labels you'll have to add the locale code to the following files to make this work:

- `themes/src/main/resources-community/theme/base/account/theme.properties`
- `themes/src/main/resources-community/theme/base/admin/theme.properties`
- `themes/src/main/resources-community/theme/base/email/theme.properties`
- `themes/src/main/resources-community/theme/base/login/theme.properties`

**Comment by jonkoops** [comment]
@sharifrezvani just checking, are you still interested in landing this?

**Comment by sharifrezvani** [comment]
@jonkoops I'm terribly sorry for replying this late. I got caught up with work as well. Yes, I will give this the final touches that you mentioned ASAP. thank you for your patience

**Comment by ssilvert** [comment]
@sharifrezvani Please do finish this up.  We are trying to get rid of our older PRs.  If you are unable to finish it in the next couple of days we can close the PR and reopen when you are ready.

Also, I'm wondering why some messages were left untranslated?

**Comment by sharifrezvani** [comment]
@ssilvert sorry again for the late reply,I hope that didn't come across as rude.

I made the changes which @jonkoops asked for.

As for the messages which are not translated, please show me some examples. I did find 4 messages which I had missed during the translation and I changed them to Farsi in my latest commit.
But if you are talking about some acronyms like "URL", the thing is, even though these words have translations in Farsi, common (even pro) users aren't aware of their translations. for example, the word "URL" would be "مکان‌یاب منبع یکسان" in Farsi, but no one knows what this means. However, almost everyone knows the meaning of URL.
You might also be talking about other messages like "Google Authenticator" or "Recaptcha", again, these are specific services by well-known companies which if they are translated, they won't sound familiar to Farsi users anymore.

However, I would gladly change the translations if you want me to. I just believe that these will be more useful to Farsi users. But if you guys have any sort of conventions, I will change them.

Still, do show me some examples so that I can fix the issues or explain them.

**Comment by jonkoops** [comment]
@sharifrezvani can you add `locale_fa` to `themes/src/main/resources/theme/base/account/messages/messages_X.properties` (X being each language) as well? That way it will show up in the list of languages when logging in, right now it's not displayed:

image

**Comment by jonkoops** [comment]
Also, the files for the Admin UI are not yet translated. Do you intend to pick that up separately?

**Comment by sharifrezvani** [comment]
@jonkoops I added the locale_fa in other language files in the commit that I just made.

And yes I'll do the Admin UI later separately. I'll open up another PR in the coming days for that.

**Comment by jonkoops** [comment]
@sharifrezvani why are those strings encoded in unicode? Doesn't UTF-8 natively support the Persian alphabet ?

**Comment by sharifrezvani** [comment]
@jonkoops it does, but the other languages aren't utf-8 so i thought that would cause a problem

**Comment by jonkoops** [comment]
Hmmm, yeah I see that it's not the case for all languages. We should fix that, so I've logged #20848 for it.

**Review by jonkoops — DISMISSED** [review]
The changes here look good, but I would like a native speaker to review this as well.

**Comment by jonkoops** [comment]
@kasir-barati would you be willing to review this pull request?

File: js/apps/account-ui/public/locales/fa/translation.json [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    "errorSignOutMessage": "خروج ناموفق بود: {{error}}",
+    "expires": "منقضی می‌شود",
+    "filterByName": "فیلتر بر اساس نام ...",
+    "firstName": "نام",
+    "fullName": "{{givenName}} {{familyName}}",
+    "groupDescriptionLabel": "View groups that you are associated with",
+    "groupLabel": "گروه‌ها",
+    "groups": "گروه‌ها",
+    "hasAccessTo": "دسترسی دارد به",
+    "infoMessage": "By clicking Remove Access, you will remove granted permissions of this application. This application will no longer use your information.",
```
Comment (ssilvert) [inline js/apps/account-ui/public/locales/fa/translation.json:63]: Not translated?

File: js/apps/account-ui/public/locales/fa/translation.json [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    "personalInfo": "اطلاعات شخصی",
+    "personalInfoDescription": "اطلاعت پایه خود را مدیریت کنید",
+    "privacyPolicy": "سیاست حریم شخصی",
+    "refreshPage": "بارگذاری مجدد صفحه",
+    "removeButton": "پاک کردن دسترسی",
+    "removeConsentError": "Could not remove consent due to: {{error}}",
+    "removeConsentSuccess": "Successfully removed consent",
+    "removeCred": "پاک کردن {{0}}",
+    "removeCredAriaLabel": "پاک کردن اعتبارنامه",
+    "removeModalMessage": "This will remove the currently granted access permission for {{0}}. You will need to grant access again if you want to use this app.",
```
Comment (ssilvert) [inline js/apps/account-ui/public/locales/fa/translation.json:103]: Not translated?

File: js/apps/account-ui/public/locales/fa/translation.json [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    "removeModalTitle": "Remove access",
```
Comment (ssilvert) [inline js/apps/account-ui/public/locales/fa/translation.json:104]: Not translated?

File: js/apps/account-ui/public/locales/fa/translation.json [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    "shareUser": "Add users to share your resource with",
+    "shareWith": "اشتراک گذاری با ",
+    "signedInDevices": "دستگاه‌های وارد شده",
+    "signedInDevicesExplanation": "Sign out of any unfamiliar devices.",
+    "signedOutSession": "خروج موفق {{0}}/{{1}}",
+    "signingIn": "در حال ورود",
+    "signingInDescription": "Configure ways to sign in.",
+    "signOut": "خروج",
+    "signOutAllDevices": "خروج از تمام دستگاه‌ها",
+    "signOutAllDevicesWarning": "This action will sign out all the devices that have signed in to your account, including the current device you are using.",
```
Comment (ssilvert) [inline js/apps/account-ui/public/locales/fa/translation.json:130]: Not translated?

File: js/apps/account-ui/public/locales/fa/translation.json [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    "shareError": "Could not share the resource due to: {{error}}",
+    "shareSuccess": "Resource successfully shared.",
+    "shareTheResource": "Share the resource - {{0}}",
+    "shareUser": "Add users to share your resource with",
+    "shareWith": "اشتراک گذاری با ",
+    "signedInDevices": "دستگاه‌های وارد شده",
+    "signedInDevicesExplanation": "Sign out of any unfamiliar devices.",
+    "signedOutSession": "خروج موفق {{0}}/{{1}}",
+    "signingIn": "در حال ورود",
+    "signingInDescription": "Configure ways to sign in.",
```
Comment (ssilvert) [inline js/apps/account-ui/public/locales/fa/translation.json:127]: Not translated?

File: js/apps/account-ui/public/locales/fa/translation.json [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    "setUpNew": "Set up {{0}}",
+    "share": "اشتراک‌گذاری",
+    "sharedWithMe": "اشتراک‌گذاری شده با من",
+    "shareError": "Could not share the resource due to: {{error}}",
+    "shareSuccess": "Resource successfully shared.",
+    "shareTheResource": "Share the resource - {{0}}",
+    "shareUser": "Add users to share your resource with",
+    "shareWith": "اشتراک گذاری با ",
+    "signedInDevices": "دستگاه‌های وارد شده",
+    "signedInDevicesExplanation": "Sign out of any unfamiliar devices.",
```
Comment (ssilvert) [inline js/apps/account-ui/public/locales/fa/translation.json:124]: Not translated?

File: js/apps/account-ui/public/locales/fa/translation.json [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    "resourceSharedWith_other": "Resource is shared with <0>{{username}}</0> and <1>{{other}}</1> other users",
+    "resourceSharedWith_zero": "This resource is not shared.",
+    "selectOne": "Select an option",
+    "setUpNew": "Set up {{0}}",
+    "share": "اشتراک‌گذاری",
+    "sharedWithMe": "اشتراک‌گذاری شده با من",
+    "shareError": "Could not share the resource due to: {{error}}",
+    "shareSuccess": "Resource successfully shared.",
+    "shareTheResource": "Share the resource - {{0}}",
+    "shareUser": "Add users to share your resource with",
```
Comment (ssilvert) [inline js/apps/account-ui/public/locales/fa/translation.json:121]: Not translated?

File: js/apps/account-ui/public/locales/fa/translation.json [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+    "required": "الزامی",
+    "resourceAlreadyShared": "منبع قبلا با این کاربر به اشتراک گذاشته شده است",
+    "resourceIntroMessage": "منابع خود را در بین اعضای تیم به اشتراک بگذارید",
+    "resourceName": "نام منبع",
+    "resources": "منابع",
+    "resourceSharedWith_one": "Resource is shared with <0>{{username}}</0>",
+    "resourceSharedWith_other": "Resource is shared with <0>{{username}}</0> and <1>{{other}}</1> other users",
+    "resourceSharedWith_zero": "This resource is not shared.",
+    "selectOne": "Select an option",
+    "setUpNew": "Set up {{0}}",
```
Comment (ssilvert) [inline js/apps/account-ui/public/locales/fa/translation.json:115]: Not translated?

**Comment by ssilvert** [comment]
> @ssilvert sorry again for the late reply,I hope that didn't come across as rude.
> 
No problem at all.  On the UI team, we try to keep older, open PRs to a minimum.   And it's not a problem if we need to close a PR and revisit later.

> As for the messages which are not translated, please show me some examples.
> 
I commented on the messages that are untranslated.  If there is some reason for leaving it untranslated then that's OK.

**Comment by sharifrezvani** [comment]
@ssilvert My mistake, I had completely forgotten about these. Thank you for pointing these out. If you'll allow me, I'll translate these in the next couple of hours. but if not, I'll do them in the next PR which I will open up for the admin-ui files. I'm asking because I don't know if it's OK for me to make any further changes at this stage.

**Comment by ssilvert** [comment]
> @ssilvert My mistake, I had completely forgotten about these. Thank you for pointing these out. If you'll allow me, I'll translate these in the next couple of hours. but if not, I'll do them in the next PR which I will open up for the admin-ui files. I'm asking because I don't know if it's OK for me to make any further changes at this stage.

@sharifrezvani If you can do it in the next couple of days then go ahead and do them in this PR.

**Comment by sharifrezvani** [comment]
@ssilvert I added the missing translations in my latest commit. thank you again for pointing them out

**Comment by jonkoops** [comment]
@kasir-barati do you have time to review the latest changes on this PR?

**Comment by ssilvert** [comment]
@sharifrezvani @kasir-barati @jonkoops I think this PR has lived long enough and a good effort has been made to get everything right.  

@kasir-barati if you want to review and make suggested changes, please go ahead.  Otherwise, I'm going to merge this as-is in 24 hours.

**Comment by jonkoops** [comment]
> But if they are busy I can suggest some other users who might be interested in reviewing this PR as a native Farsi speaker

@sharifrezvani Yes, feel free to ping anyone you might know who's interested.

File: themes/src/main/resources-community/theme/keycloak.v2/account/messages/messages_fa.properties [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+inUse=در حال استفاده
+notInUse=آزاد
+applicationDetails=جزئیات اپلیکیشن
+client=مشتری
+description=توضیحات
+baseUrl=URL
+accessGrantedOn=دسترسی داده شد در
+removeButton=حذف دسترسی
+removeModalTitle=حذف دسترسی
+removeModalMessage=این عمل مجوز دسترسی اعطاشده فعلی برای {0} را حذف می‌کند. اگر می‌خواهید از این اپ دوباره استفاده کنید باید مجدد دسترسی را اعطا کنید.
```
Comment (kasir-barati) [inline themes/src/main/resources-community/theme/keycloak.v2/account/messages/messages_fa.properties:136]:
```suggestion
removeModalMessage=با این کار مجوز دسترسی اعطا شده در حال حاضر برای {0} حذف خواهد شد. اگر می خواهید از این اپلیکیشن استفاده کنید، باید دوباره مجوز دسترسی بدهید.
```

Using the same translated message here as we had in this file: [js/apps/account-ui/public/locales/fa/translation.json](https://github.com/keycloak/keycloak/pull/19747/files#diff-572c71ad3aafccaceadd91c102a082fe63becb83163d878ee158dc11837c8b64)

**Review by kasir-barati — CHANGES_REQUESTED** [review]
Hi dear reader

I just found some typoes and also inconsistencies.
And my apologize to @jonkoops @sharifrezvani for late reply :sweat:

**Comment by ssilvert** [comment]
@kasir-barati I'm sorry.  I merged this without seeing your suggested changes.  If you feel they are important enough, please submit a new PR.

**Pull request merged**